### PR TITLE
Custom headers per function

### PR DESCRIPTION
**Pull request by jensdev**
This PR already includes #306 

All functions should have a last optional options object parameter which can include custom headers

### CODE REVIEW

**Comment by ThomasReyskens** [comment]
make options optional again in implementation

File: src/services/files/types.ts [hunk 1]
```diff
@@ -42,30 +47,15 @@ export interface FilesService {
...
-   * Find First
-   * @param rql an optional rql string
-   * @returns the first element found
-   */
-  findFirst(
-    this: FilesService,
-    options?: { rql?: RQLString }
-  ): Promise<FileDetails>;
+  findByName(name: string, options?: OptionsWithRql): Promise<FileDetails>;
+  findFirst(options?: OptionsWithRql): Promise<FileDetails>;
```
Comment (ThomasReyskens) [inline src/services/files/types.ts:58]: why is the find first jsdoc gone?

**Review by ThomasReyskens — CHANGES_REQUESTED** [review]
open comments

File: src/services/dispatchers/dispatchers.ts [hunk 2]
```diff
@@ -1,31 +1,41 @@
...
-    return (await client.post(httpAuth, '/', requestBody)).data;
+  async create(requestBody, options) {
+    return (await client.post(httpAuth, '/', requestBody, options)).data;
   },
 
-  async remove(dispatcherId: ObjectId): Promise<AffectedRecords> {
-    return (await client.delete(httpAuth, `/${dispatcherId}`)).data;
+  /**
+   * Delete a dispatcher
+   * Permission | Scope | Effect
```
Comment (ThomasReyskens) [inline src/services/dispatchers/dispatchers.ts:31]: add newline before permissions, to create a table in jsdoc

File: src/services/auth/types.ts [hunk 3]
```diff
@@ -137,3 +144,183 @@ export interface MfaMethodVerification {
...
+   * @async
+   * @see https://developers.extrahorizon.io/swagger-ui/?url=https://developers.extrahorizon.io/services/auth-service/2.0.4-dev/openapi.yaml#/Applications/post_applications
+   */
+  create(
+    data: ApplicationCreation,
+    options?: OptionsBase
+  ): Promise<Application>;
+  /**
+   * Get a list of applications
+   * @permission VIEW_APPLICATIONS | scope:global |
```
Comment (ThomasReyskens) [inline src/services/auth/types.ts:162]: newline before permission, to be consistent

File: src/services/files/types.ts (hunk 1 above)
Comment (jensdev) [reply src/services/files/types.ts:58]: I've added it

**Review by ThomasReyskens — APPROVED** [review]

**Pull request merged**